module/userProjects/helper.js

```diff
@@ -2266,6 +2298,27 @@ module.exports = class UserProjectsHelper {
                         programAndSolutionInformation.data
                     )
                 }
+
+                let programJoined = await programUsers.findProgramJoined(requestedData.programId,userId)
+                if(programJoined.length == 0){
+                   
+                    let programJoinBody = {};
+                    programJoinBody.userRoleInformation = requestedData.userRoleInformation ? requestedData.userRoleInformation : {};
```

Comment:
@ankitpws if importFromLibrary function is used to create a project for a nontargeted user, we can avoid this function because we won't be adding private program to programUsers